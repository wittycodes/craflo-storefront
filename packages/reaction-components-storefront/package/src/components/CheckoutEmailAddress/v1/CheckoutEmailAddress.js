import React, { Component } from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { addTypographyStyles, applyTheme } from "../../../utils";

const StyledDiv = styled.div`
  ${addTypographyStyles("CheckoutEmailAddress", "labelText")}
  border-bottom-color: ${applyTheme("CheckoutEmailAddress.borderBottomColor")};
  border-bottom-style: solid;
  border-bottom-width: ${applyTheme("CheckoutEmailAddress.borderBottomWidth")};
  border-left-color: ${applyTheme("CheckoutEmailAddress.borderLeftColor")};
  border-left-style: solid;
  border-left-width: ${applyTheme("CheckoutEmailAddress.borderLeftWidth")};
  border-right-color: ${applyTheme("CheckoutEmailAddress.borderRightColor")};
  border-right-style: solid;
  border-right-width: ${applyTheme("CheckoutEmailAddress.borderRightWidth")};
  border-top-color: ${applyTheme("CheckoutEmailAddress.borderTopColor")};
  border-top-style: solid;
  border-top-width: ${applyTheme("CheckoutEmailAddress.borderTopWidth")};
  padding-bottom: ${applyTheme("CheckoutEmailAddress.paddingBottom")};
  padding-left: ${applyTheme("CheckoutEmailAddress.paddingLeft")};
  padding-right: ${applyTheme("CheckoutEmailAddress.paddingRight")};
  padding-top: ${applyTheme("CheckoutEmailAddress.paddingTop")};
`;

const StyledSpan = styled.span`
  ${addTypographyStyles("CheckoutEmailAddressEmphasis", "labelTextBold")}
`;

class CheckoutEmailAddress extends Component {
  static propTypes = {
    /**
     * You can provide a `className` prop that will be applied to the outermost DOM element
     * rendered by this component. We do not recommend using this for styling purposes, but
     * it can be useful as a selector in some situations.
     */
    className: PropTypes.string,
    emailAddress: PropTypes.string.isRequired,
    isAccountEmail: PropTypes.bool.isRequired,
    /**
     * The text for the "Signed in as" label text.
     */
    signedInText: PropTypes.string
  };

  static defaultProps = {
    signedInText: "Signed in as"
  }

  renderAccountEmail = () => {
    const { isAccountEmail, signedInText } = this.props;

    if (isAccountEmail) {
      return signedInText;
    }

    return null;
  };

  render() {
    const { className, emailAddress } = this.props;
    return (
      <StyledDiv className={className}>
        {this.renderAccountEmail()} <StyledSpan>{emailAddress}</StyledSpan>
      </StyledDiv>
    );
  }
}

export default CheckoutEmailAddress;
