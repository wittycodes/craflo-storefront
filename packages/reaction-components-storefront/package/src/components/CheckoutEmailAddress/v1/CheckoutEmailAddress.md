### Overview
Displays a customers email address inside the checkout flow.

### Usage

#### Account user

```jsx
<CheckoutEmailAddress
  emailAddress="account.email@example.com"
  isAccountEmail={true}
/>
```

#### Guest user

```jsx
<CheckoutEmailAddress
  emailAddress="guest.email@example.com"
  isAccountEmail={false}
/>
```

### Theme

Assume that any theme prop that does not begin with "rui" is within `rui_components`. See [Theming Components](./#!/Theming%20Components).

| Theme Prop                                         | Default | Description                                                                             |
| -------------------------------------------------- | ------- | --------------------------------------------------------------------------------------- |
| `borderBottomColor` | black10 | Color of bottom border |
| `borderBottomWidth` | 2px | Width of bottom border |
| `borderLeftColor` | black10 | Color of left border |
| `borderLeftWidth` | 0 | Width of left border |
| `borderRightColor` | black10 | Color of right border |
| `borderRightWidth` | 0 | Width of right border |
| `borderTopColor` | black10 | Color of top border |
| `borderTopWidth` | 0 | Width of top border |
| `paddingBottom` | 16px | Bottom padding |
| `paddingLeft` | 0 | Left padding |
| `paddingRight` | 0 | Right padding |
| `paddingTop` | 16px | Top padding |

#### Typography

- The "Signed in as" text uses `labelText` style with `rui_components.CheckoutEmailAddress` override
- The email address text uses `labelTextBold` style with `rui_components.CheckoutEmailAddressEmphasis` override
