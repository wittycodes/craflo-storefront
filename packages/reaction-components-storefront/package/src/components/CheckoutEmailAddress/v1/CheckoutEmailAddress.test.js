import React from "react";
import renderer from "react-test-renderer";
import CheckoutEmailAddress from "./CheckoutEmailAddress";

test("render email address of a user with an account", () => {
  const component = renderer.create((
    <CheckoutEmailAddress
      emailAddress="account.email@example.com"
      isAccountEmail={true}
    />
  ));

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("render email address of a guest", () => {
  const component = renderer.create((
    <CheckoutEmailAddress
      emailAddress="guest.email@example.com"
      isAccountEmail={false}
    />
  ));

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
