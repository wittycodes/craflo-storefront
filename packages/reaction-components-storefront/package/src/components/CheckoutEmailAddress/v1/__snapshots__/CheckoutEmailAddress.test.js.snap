// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render email address of a guest 1`] = `
.c0 {
  -webkit-font-smoothing: antialiased;
  color: #3c3c3c;
  font-family: 'Source Sans Pro','Helvetica Neue',Helvetica,sans-serif;
  font-size: 14px;
  font-style: normal;
  font-stretch: normal;
  font-weight: 400;
  -webkit-letter-spacing: .02em;
  -moz-letter-spacing: .02em;
  -ms-letter-spacing: .02em;
  letter-spacing: .02em;
  line-height: 1.25;
  border-bottom-color: #e6e6e6;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-left-color: #e6e6e6;
  border-left-style: solid;
  border-left-width: 0;
  border-right-color: #e6e6e6;
  border-right-style: solid;
  border-right-width: 0;
  border-top-color: #e6e6e6;
  border-top-style: solid;
  border-top-width: 0;
  padding-bottom: 16px;
  padding-left: 0;
  padding-right: 0;
  padding-top: 16px;
}

.c1 {
  -webkit-font-smoothing: antialiased;
  color: #505558;
  font-family: 'Source Sans Pro','Helvetica Neue',Helvetica,sans-serif;
  font-size: 14px;
  font-style: normal;
  font-stretch: normal;
  font-weight: 700;
  -webkit-letter-spacing: .02em;
  -moz-letter-spacing: .02em;
  -ms-letter-spacing: .02em;
  letter-spacing: .02em;
  line-height: 1.25;
}

<div
  className="c0"
>
   
  <span
    className="c1"
  >
    guest.email@example.com
  </span>
</div>
`;

exports[`render email address of a user with an account 1`] = `
.c0 {
  -webkit-font-smoothing: antialiased;
  color: #3c3c3c;
  font-family: 'Source Sans Pro','Helvetica Neue',Helvetica,sans-serif;
  font-size: 14px;
  font-style: normal;
  font-stretch: normal;
  font-weight: 400;
  -webkit-letter-spacing: .02em;
  -moz-letter-spacing: .02em;
  -ms-letter-spacing: .02em;
  letter-spacing: .02em;
  line-height: 1.25;
  border-bottom-color: #e6e6e6;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-left-color: #e6e6e6;
  border-left-style: solid;
  border-left-width: 0;
  border-right-color: #e6e6e6;
  border-right-style: solid;
  border-right-width: 0;
  border-top-color: #e6e6e6;
  border-top-style: solid;
  border-top-width: 0;
  padding-bottom: 16px;
  padding-left: 0;
  padding-right: 0;
  padding-top: 16px;
}

.c1 {
  -webkit-font-smoothing: antialiased;
  color: #505558;
  font-family: 'Source Sans Pro','Helvetica Neue',Helvetica,sans-serif;
  font-size: 14px;
  font-style: normal;
  font-stretch: normal;
  font-weight: 700;
  -webkit-letter-spacing: .02em;
  -moz-letter-spacing: .02em;
  -ms-letter-spacing: .02em;
  letter-spacing: .02em;
  line-height: 1.25;
}

<div
  className="c0"
>
  Signed in as
   
  <span
    className="c1"
  >
    account.email@example.com
  </span>
</div>
`;
